### PR TITLE
[macOS] Software updates week 8 2021

### PR DESCRIPTION
**Pull request by LeonidLapshin**
# Description

Default Xcode changed to 12.4, and default Xamarin bundle changed to 6_12_6 which includes Xamarin.iOS 14.10, Xamarin.Android 11.1, Xamarin.Mac 7.4, Mono 6.12

Mono updated to 6.12.0.122

## Check list
- [ ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Review by dibir-magomedsaygitov — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

**Comment by miketimofeev** [comment]
@LeonidLapshin we need to change the default Xcode to 12.4 and the default xamarin bundle to 6_12_6 for 10.15 & 11.0
https://github.com/actions/virtual-environments/issues/2685

**Review by al-cheb — APPROVED** [review]

**Review by miketimofeev — CHANGES_REQUESTED** [review]
Please change default versions

**Review by miketimofeev — APPROVED** [review]

File: images/macos/toolsets/toolset-10.15.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "xamarin": {
         "vsmac": "latest",
         "mono-versions": [
-            "6.12.0.113", "6.10.0.106", "6.8.0.123", "6.6.0.166", "6.4.0.208"
```
Comment (AlenaSviridenko) [inline images/macos/toolsets/toolset-10.15.json:23]: no need to update patch version of mono, we add new minor versions only

Comment (LeonidLapshin) [reply images/macos/toolsets/toolset-10.15.json]: done, thank you!

**Review by maxim-lobanov — APPROVED** [review]

**Pull request merged**